### PR TITLE
enhancement: Show agent availability status in conversation assignee select list

### LINKED ISSUE #1892: Enhancement: Display agent status on assignation.
**Is your enhancement request related to a problem? Please describe.**  
Whenever an agent wants to assign somebody else in a conversation it is kind of tedious as they don't know instantly who is available. (See screenshot).  

**Describe the solution you'd like**  
Categorize the agents you can assign to with "Online" agents first, then "Busy" and finally "Offline" ones.  

**Describe alternatives you've considered**  
- Display the agent status next to their name.  
- Allow a filter that each agent can set as they want (only see online people, for example).  

**Additional context**   
Screenshot 2021-03-10 at 14 04 58
In the above screenshot I don't know if "Kevin" is available, busy or offline.

### PR DESCRIPTION
**Pull request by muhsin-k**
Fixes #1892 

**Preview**

**conversation header**

photo_2021-04-19 11 34 45

**contact panel**

Screenshot 2021-04-21 at 10 13 58 AM

### CODE REVIEW

**Review by nithindavid — APPROVED** [review]

**Comment by nithindavid** [comment]
@muhsin-k Does this cover the Agent dropdown on sidebar ?

**Comment by muhsin-k** [comment]
> @muhsin-k Does this cover the Agent dropdown on sidebar ?

Yes. Created a new component **AvailabilityStatusBadge** and used it everywhere.

File: app/javascript/dashboard/components/widgets/conversation/ConversationHeader.vue [hunk 1]
```diff
@@ -142,4 +151,23 @@ export default {
 .conv-header {
   flex: 0 0 var(--space-jumbo);
 }
+.status-badge {
```
Comment (nithindavid) [inline app/javascript/dashboard/components/widgets/conversation/ConversationHeader.vue:154]: Can you remove this ?

File: app/javascript/dashboard/routes/dashboard/conversation/ContactPanel.vue [hunk 2]
```diff
@@ -342,4 +354,20 @@ export default {
 .multiselect__label {
   margin-bottom: var(--space-smaller);
 }
+.status-badge {
```
Comment (nithindavid) [inline app/javascript/dashboard/routes/dashboard/conversation/ContactPanel.vue:357]: This one also

**Review by nithindavid — CHANGES_REQUESTED** [review]

**Review by nithindavid — CHANGES_REQUESTED** [review]
Please see comments

**Comment by muhsin-k** [comment]
@nithindavid Please check the lastest changes

**Review by nithindavid — APPROVED** [review]

**Pull request merged**